README.md

```diff
@@ -183,6 +189,20 @@ services:
         # that should be set in the context of the process
         environment:
             <env var name>: <env var value>
+
+        # (Optional) User ID for starting service as a different user. Takes
+        # precedence over "user" field.
+        user-id: <uid>
+
+        # (Optional) Username for starting service as a different user.
+        user: <username>
+
+        # (Optional) Group ID for starting service as a different user. Takes
+        # precedence over "group" field.
+        group-id: <gid>
+
+        # (Optional) Group name or GID for starting service as a different user.
+        group: <group name>
```

Comment:
I wonder if we just treat it as an error to set both group-id and group and have them not match in the target container.

Reply:
I think that behavior is probably a better choice -- however, this function is already used in the files API to normalize the user / user-id combination (which is in the wild already) so strictly speaking we can't change this behavior. We could make this usage of user/user-id behave differently, but that seems weird. So it's not perfect, but precedence does seem like a fairly common and easy-to-understand concept, so not too bad either.

Reply:
The behavior I remember discussing in the context of the operator was precisely to conflict if both are provided until we had a clear idea, but maybe my memory fails. It seems awkward for the ID to have precedence instead of the name, because the name is more indicative of what the user is. Let's just ponder for a moment: imagine we have user=postgres/uid=400, and 400 is serial-port. Does it make sense to make the postgres files owned by serial-port? Feels like it doesn't. If there is a precedence, it's probably the name that makes more sense than the ID. I'd prefer to fix that now than to make the weird behavior permanent.

Reply:
Fair enough - I've made this change now (and this function is both used here and for the files API). As far as backwards compatibility goes, it seems *very* unlikely with the number of charms we have so far -- and even fewer use the files API -- that someone would be specifying both in a conflicting way. So on reflection I do think it's fine to make this minor breaking change.

Reply:
I'm happy with a "these differ, therefore error"

Reply:
Thanks, looks good.